components/dashboard/src/service/json-rpc-workspace-client.ts

```diff
@@ -160,4 +171,115 @@ export class JsonRpcWorkspaceClient implements PromiseClient<typeof WorkspaceSer
         result.workspace = workspace.workspace;
         return result;
     }
+
+    async stopWorkspace(
+        request: PartialMessage<StopWorkspaceRequest>,
+        _options?: CallOptions | undefined,
+    ): Promise<StopWorkspaceResponse> {
+        if (!request.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        await getGitpodService().server.stopWorkspace(request.workspaceId);
+        const result = new StopWorkspaceResponse();
+        return result;
+    }
+
+    async deleteWorkspace(
+        request: PartialMessage<DeleteWorkspaceRequest>,
+        _options?: CallOptions | undefined,
+    ): Promise<DeleteWorkspaceResponse> {
+        if (!request.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        await getGitpodService().server.deleteWorkspace(request.workspaceId);
+        const result = new DeleteWorkspaceResponse();
+        return result;
+    }
+
+    async updateWorkspace(
+        request: PartialMessage<UpdateWorkspaceRequest>,
+        _options?: CallOptions | undefined,
+    ): Promise<UpdateWorkspaceResponse> {
+        if (!request.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+
+        // check if user can access workspace first
+        await this.getWorkspace({ workspaceId: request.workspaceId });
```

Comment:
Clients don't need to do such checks. This is entirely the responsibility of the server

Reply:
See https://github.com/gitpod-io/gitpod/pull/19118#discussion_r1404074299